src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+/**
+ * A super-class to Address, Email and Phone classes
+ */
+public abstract class Contact {
```

Comment:
Very good. Making it abstract can help to ensure no "Contact" object can be created.